### PR TITLE
openstack: Update base images

### PR DESCRIPTION
**Pull request by pierreprinetti**
* Update OpenStack CI base images to match the Installer's
* Use the OpenStack repositories embedded in the base image
* Remove the pinning of the openstackSDK library, as the bug that required it is marked as resolved.

### CODE REVIEW

**Review by mandre — COMMENTED** [review]
For some reason it's not using the `OpenStack Yoga Repository` you're providing, and it's still pulling openstack rpms from `openstack-16-for-rhel-8-rpms`. Not sure where that repo is coming from.
Interestingly, the `rdo-train` repo wasn't used at all. From [a different CI job](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_installer/6083/pull-ci-openshift-installer-master-images/1544507474105602048/artifacts/build-logs/openstack-installer.log):
```
2022-07-06T03:01:59.568022776Z Errors during downloading metadata for repository 'openstack-train':
2022-07-06T03:01:59.568022776Z   - Status code: 404 for http://mirror.centos.org/centos/8/cloud/x86_64/openstack-train/repodata/repomd.xml (IP: 67.220.193.10)
2022-07-06T03:01:59.568022776Z Error: Failed to download metadata for repo 'openstack-train': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
```

**Review by dulek — CHANGES_REQUESTED** [review]

File: images/openstack/Dockerfile.ci [hunk 1]
```diff
@@ -1,14 +1,14 @@
 # This Dockerfile is used by CI to test using OpenShift Installer against an OpenStack cloud.
 # It builds an image containing the openshift-install command as well as the openstack cli.
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.10 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.11 AS builder
```
Comment (mandre) [inline images/openstack/Dockerfile.ci:3]: registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.12 is available now, shouldn't we use it?

**Review by mandre — COMMENTED** [review]
/lgtm
/approve
/hold for CI

**Pull request merged**